devnotes/tpm.md

```diff
@@ -0,0 +1,205 @@
+This document is an attempt to gather information about the issues faced with
+TPMs on POWER9 and what are the potential solutions to them.
+
+## Some definitions
+
+* fTPM - Firmware-based TPM implementation
+
+## Problems
+
+### POWER9 cannot generate LPC TPM cycles
+
+This is a hardware limitation and cannot be worked around in firmware.
+
+The consequence is that regular LPC TPM cannot be used on POWER9 by directly
+connecting it to the TPM LPC header (one might wonder why have the LPC TPM
+header in the first place...).
+
+## Quick overview
+
+| Aspect\Approach | [I2C]   |[On BMC]|[FPGA LPC]|[FPGA SPI]|[On MCU] |[LibreBMC]| [SBE] |
```

Comment:
@SergiiDmytruk Can we try to work on assigning some "points" to have a scale comparing different aspects of these solutions? 

@TomaszAIR We did something like that for Fobnail research. Could you please share the idea and link here?

Reply:
Sure.

@SergiiDmytruk please look at this document. https://github.com/fobnail/docs/blob/minimal-os-zephyr%2Bsel4-research/dev-notes/minimal-os-for-fobnail.md#summary In the summary there is a table which present `score` for every analyzed idea/solution. Here are the considered functionalities https://github.com/fobnail/docs/blob/minimal-os-zephyr%2Bsel4-research/dev-notes/minimal-os-for-fobnail.md#os-requirements and here explaination how those points were added https://github.com/fobnail/docs/blob/minimal-os-zephyr%2Bsel4-research/dev-notes/minimal-os-for-fobnail.md#os-score-counting